presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/StaticHiveCluster.java

```diff
@@ -45,10 +45,12 @@ public StaticHiveCluster(List<URI> metastoreUris, String metastoreUsername, Hive
     {
         requireNonNull(metastoreUris, "metastoreUris is null");
         checkArgument(!metastoreUris.isEmpty(), "metastoreUris must specify at least one URI");
-        this.addresses = metastoreUris.stream()
+
+        this.addresses = new LinkedList<>();
```

Comment:
It you still want to shuffle list (see: @kokosing comment) this should be `ArrayList`.